src/ralph/lib/polymorphic/models.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+
+from itertools import groupby
+
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+from django.db.models.fields.related import OneToOneField
+
+
+class PolymorphicQuerySet(models.QuerySet):
+
+    def iterator(self):
+        """
+        Override iterator:
+            - Iterate for all objects and collected ID
+            - For each ContentType generates additional queryset
+            - Returns iterator with different models
+        """
+        result = []
+        content_type_model_map = {}
+        for obj in super().iterator():
+            content_type_model_map[
+                obj.content_type_id
+            ] = obj.content_type.model_class()
+            result.append((
+                obj.content_type_id, obj.pk)
+            )
+        result = sorted(result, key=lambda x: x[0])
+        result = groupby(result, lambda x: x[0])
+
+        result_query = []
+        for k, v in result:
+            model = content_type_model_map[k]
+            polymorphic_models = getattr(model, '_polymorphic_models', [])
+            if polymorphic_models and model not in polymorphic_models:
+                result_query.extend(
+                    model.objects.filter(pk__in=[i[1] for i in v])
+                )
+
+        for obj in result_query:
+            yield obj
+
+
+class PolymorphicBase(models.base.ModelBase):
+    def __new__(cls, name, bases, attrs):
+        full_mro = set(
+            tuple([mro for b in bases for mro in b.__mro__]) + bases
+        )
+        base_polymorphic = set(
+            [b for b in full_mro if issubclass(b, Polymorphic)]
+        )
+        attrs['_polymorphic_models'] = base_polymorphic
```

Comment:
i'd mark here every class from `base_polymorphic` that polymorphic query should be used for it - for now it should only apply to `BaseObject` and `Asset`. Then you could use this param instead of checking `_polymorphic_models` in autocomplete.